### PR TITLE
Implement Export feature for V1 conversations with comprehensive unit tests

### PR DESCRIPTION
**Pull request by tofarr**
## Summary of PR

This PR implements the complete "Download Trajectory" feature for V1 conversations, allowing users to download conversation events and metadata as a ZIP file. The implementation includes both backend API endpoints and frontend UI components, along with comprehensive unit tests.

image

image

### Backend Implementation:

**New API Endpoint:**
- `GET /api/v1/conversations/{conversation_id}/download-trajectory` - Downloads conversation as ZIP file
- Returns ZIP containing conversation metadata (`meta.json`) and all events as individual JSON files
- Implements proper error handling for missing conversations
- Uses pagination with `iterate` function from `search_utils.py` for efficient event retrieval

**Service Layer Improvements:**
- Enhanced `LiveStatusAppConversationService.download_conversation_trajectory()` method
- Uses `conversation_info.model_dump_json()` for metadata serialization (as requested)
- Replaced while loop with `iterate` function for better pagination handling
- Added adapter pattern to work with `EventPage` objects

**Utility Enhancements:**
- Updated `search_utils.py` with improved `iterate` function for pagination

### Frontend Implementation:

**UI Components:**
- Added "Download Trajectory" option to conversation card kebab menu
- Integrated download functionality in conversation card actions
- Added proper loading states and error handling
- Implemented file download with appropriate filename generation

**API Integration:**
- New API client method in `v1-conversation-service.api.ts`
- Proper TypeScript types and error handling
- Integration with existing conversation service architecture

**Internationalization:**
- Added translation keys for "Download Trajectory" in multiple languages
- Updated i18n declaration files

### Comprehensive Unit Tests:

**5 Test Methods Added:**
1. **`test_download_conversation_trajectory_success`** - Tests successful download with multiple paginated events, ZIP file creation, and content validation
2. **`test_download_conversation_trajectory_conversation_not_found`** - Tests error handling when conversation doesn't exist
3. **`test_download_conversation_trajectory_empty_events`** - Tests behavior with conversations that have no events
4. **`test_download_conversation_trajectory_large_pagination`** - Tests pagination with multiple pages (4 pages × 3 events = 12 total)
5. **`test_download_conversation_trajectory_iterate_function_usage`** - Tests that the method correctly uses the `iterate` function

**Test Features:**
- Real ZIP file validation with content verification
- Comprehensive pagination testing
- Error handling and edge case coverage
- Service integration verification
- JSON content validation for events and metadata
- Proper async/await patterns and mocking

### Technical Improvements:

- **Pagination Efficiency**: Replaced manual while loop with reusable `iterate` function
- **Data Serialization**: Uses `model_dump_json()` for consistent JSON output
- **Error Handling**: Proper exception handling with descriptive messages
- **Code Quality**: All pre-commit hooks pass (ruff, mypy, formatting)
- **Test Coverage**: Comprehensive unit tests with 100% scenario coverage

### User Experience:

- Users can now download complete conversation trajectories for analysis
- ZIP file contains structured data with metadata and chronological events
- Accessible via intuitive kebab menu in conversation cards
- Proper loading states and error feedback

## Change Type

- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Refactor
- [ ] Other (dependency update, docs, typo fixes, etc.)

## Checklist

- [ ] I have read and reviewed the code and I understand what the code is doing.
- [ ] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes

Implements the "Download Trajectory" feature for V1 conversations as requested, including backend endpoint, frontend UI, and comprehensive unit tests.

## Release Notes

- [x] Include this change in the Release Notes.

**New Feature**: Added "Download Trajectory" functionality for V1 conversations, allowing users to download conversation events and metadata as a structured ZIP file for analysis and backup purposes.

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.openhands.dev/openhands/runtime:7342638-nikolaik   --name openhands-app-7342638   docker.openhands.dev/openhands/openhands:7342638
```

### CODE REVIEW

File: openhands/app_server/app_conversation/app_conversation_router.py [hunk 1]
```diff
@@ -400,6 +400,51 @@ async def read_conversation_file(
...
+
+    Returns a zip file containing all events and metadata for the conversation.
+
+    Args:
+        conversation_id: The UUID of the conversation to download
+
+    Returns:
+        A zip file containing the conversation trajectory
+    """
+    from fastapi.responses import Response
```
Comment (hieptl) [inline openhands/app_server/app_conversation/app_conversation_router.py]: **question:** Could we relocate this import statement to the top of the file?

File: openhands/app_server/app_conversation/app_conversation_router.py [hunk 2]
```diff
@@ -400,6 +400,51 @@ async def read_conversation_file(
...
+            headers={
+                'Content-Disposition': f'attachment; filename="conversation_{conversation_id}.zip"'
+            },
+        )
+    except ValueError as e:
+        from fastapi import HTTPException
+
+        raise HTTPException(status_code=404, detail=str(e))
+    except Exception as e:
+        from fastapi import HTTPException
```
Comment (hieptl) [inline openhands/app_server/app_conversation/app_conversation_router.py]: **question:** Could we relocate this import statement to the top of the file?

File: openhands/app_server/app_conversation/live_status_app_conversation_service.py [hunk 3]
```diff
@@ -1002,6 +1009,77 @@ async def _delete_from_database(
 
         return deleted_info or deleted_tasks
 
+    async def export_conversation(self, conversation_id: UUID) -> bytes:
```
Comment (hieptl) [inline openhands/app_server/app_conversation/live_status_app_conversation_service.py:1188]: **question:** Could this function, or part of its logic, be moved to the AppConversationServiceBase class?

**Comment by hieptl** [comment]
**question:** I have verified the pull request locally, and it functions correctly for both v0 and v1. On the conversation page for both versions, the Export Conversation option is available.

However, there is an inconsistency in behavior within the conversation panel UI when hovering over the menu icon:
- v0 conversations: The Export Conversation option is not displayed.
- v1 conversations: The Export Conversation option is displayed.

This discrepancy may lead to user confusion.

Please refer to the video below for additional details. Thank you! 🙏 

https://github.com/user-attachments/assets/5bb58bff-f53c-4b01-97ab-cbb879139c7e

**Comment by hieptl** [comment]
**question:** In v0, exporting a conversation results in a file name with the format trajectory-{conversation_id}. In v1, the exported file name follows the format conversation_{conversation_id}. Should we align v1 with v0 to ensure consistency?

File: frontend/src/components/features/conversation-panel/conversation-card/conversation-card.tsx [hunk 4]
```diff
@@ -101,6 +102,36 @@ export function ConversationCard({
...
+    event: React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    event.preventDefault();
+    event.stopPropagation();
+    posthog.capture("download_trajectory_button_clicked");
+
+    if (conversationId && conversationVersion === "V1") {
+      try {
+        const blob =
+          await V1ConversationService.downloadConversation(conversationId);
```
Comment (hieptl) [inline frontend/src/components/features/conversation-panel/conversation-card/conversation-card.tsx]: **suggestion:** Could we create a hook that uses react-query to interact with the service class? Typically, we avoid calling a service class directly within a component.

File: frontend/src/hooks/use-conversation-name-context-menu.ts [hunk 5]
```diff
@@ -149,6 +152,35 @@ export function useConversationNameContextMenu({
     onContextMenuToggle?.(false);
   };
 
+  const handleDownloadConversation = async (
+    event: React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    event.preventDefault();
+    event.stopPropagation();
+    posthog.capture("download_conversation_button_clicked");
```
Comment (hieptl) [inline frontend/src/hooks/use-conversation-name-context-menu.ts]: **suggestion:** Are we capturing different PostHog events in this component and in the conversation-card.tsx component, even though they perform the same function? Should we standardize this by capturing the same PostHog event in both cases?

File: frontend/src/hooks/use-conversation-name-context-menu.ts [hunk 6]
```diff
@@ -149,6 +152,35 @@ export function useConversationNameContextMenu({
     onContextMenuToggle?.(false);
   };
 
+  const handleDownloadConversation = async (
```
Comment (hieptl) [inline frontend/src/hooks/use-conversation-name-context-menu.ts:155]: **suggestion:** it appears that this function is duplicated between this component and the conversation-card.tsx component. We could create a shared hook that exports this function, making it reusable across both files.

File: openhands/app_server/app_conversation/app_conversation_router.py (hunk 1 above)
Comment (tofarr) [reply openhands/app_server/app_conversation/app_conversation_router.py]: I hadn't noticed that the LLM did this. Weird.

File: frontend/src/hooks/use-conversation-name-context-menu.ts (hunk 5 above)
Comment (tofarr) [reply frontend/src/hooks/use-conversation-name-context-menu.ts]: Good call - using same event now

File: frontend/src/hooks/use-conversation-name-context-menu.ts (hunk 6 above)
Comment (tofarr) [reply frontend/src/hooks/use-conversation-name-context-menu.ts:155]: Done

**Comment by tofarr** [comment]
> **question:** In v0, exporting a conversation results in a file name with the format trajectory-{conversation_id}. In v1, the exported file name follows the format conversation_{conversation_id}. Should we align v1 with v0 to ensure consistency?

I actually think we should steer away from the term "trajectory", and just call it "Download Conversation". I did not update the label in the UI yet for consistency.

File: frontend/src/components/features/conversation-panel/conversation-card/conversation-card.tsx (hunk 4 above)
Comment (tofarr) [reply frontend/src/components/features/conversation-panel/conversation-card/conversation-card.tsx]: I added a utility function for this handler.

File: openhands/app_server/app_conversation/live_status_app_conversation_service.py (hunk 3 above)
Comment (tofarr) [reply openhands/app_server/app_conversation/live_status_app_conversation_service.py:1188]: The function grabs app_conversation_info which does not include the status of the pod, as we don't need it. There is no way of doing this in the base class.

It's a small thing overall.

**Review by hieptl — APPROVED** [review]
Awesome! 🎉  Thank you! 🙏
